### DISCUSSION TITLE
Inferring typescript types based off collections?

### DISCUSSION THREAD

**Discussion by omawhite** [StaticJsCMS/static-cms#998] [Q&A]
I was wondering if anyone knew of a way to easily infer types from your collections? Currently I'm doing it by hand like so:
```typescript

// Read markdown file as string
const fullPath = path.join(postsDirectory, fileName);
const fileContents = fs.readFileSync(fullPath, 'utf8');
const matterResult = matter(fileContents);
const matterData = matterResult.data as {
      date: string;
      title: string;
      lastUpdated?: string;
      tags?: string[];
    };
 ```
But it would be nice if I could figure out a way from my code to just infer the types somehow. Does anyone have any suggestions.

**Comment by GlenLamb** [comment]
https://www.staticcms.org/docs/typescript
